src/Chapter2.hs

```diff
@@ -841,7 +861,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n xs 
+    |n < 0 = []
+    |otherwise = subList n to (cycle xs)
```

Comment:
Oh, cool reusage of `sublist` in here 👍🏼 

Additional trick you can use in here to speed up the solution is to take `mod` by the length of the list for n, In this case, you won't need to go extra cycles of the same list 🙂  (But note that it only works with non-empty lists)